jwst/associations/lib/rules_level3.py

```diff
@@ -277,6 +275,22 @@ def __init__(self, *args, **kwargs):
                     ),
                 ],
                 reduce=Constraint.notany
+            ),
+            # Constaint to prevent calibration data from level 3 processing
+            Constraint(
+                [
+                    DMSAttrConstraint(
+                        name='restricted_slitless',
+                        sources=['exp_type'],
+                        value = ('mir_lrs-slitless')
+                    ),
+                    DMSAttrConstraint(
+                        name='tso_obs',
+                        sources=['tso_visit'],
+                        value = ('T')
+                    ),
+                ],
```

Comment:
So, the effect of these constraints is to only allow exposures with both EXP_TYPE='MIR_LRS-SLITLESS' and TSO_VISIT=True to be included in a level-3 ASN? If so, that seems to do the trick.

Reply:
That is correct. Observation 27 has an entry with EXP_TYPE=MIR_LRS-SLITLESS &  TSO_VISIT=T and level 3 products are created,  jw00623-o027_tso3_001_asn.json